### PR TITLE
Fix: check build for arm architecture

### PR DESCRIPTION
**Pull request by AkashKumar7902**
## Related Issue
  - Info about Issue or bug

Closes: #986 

#### Describe the changes you've made
support for architecture came in [#253](https://github.com/actions/setup-go/pull/253)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by charankamarapu** [comment]
We need both amd(which was present by default) as well as arm. You can check PR-https://github.com/keploy/keploy/pull/985 and for Github action I have deleted there will be corresponding Github action for amd , Like go.yml, main.yml, sample-run.yml. By default they run on amd you should also have arm check for it.

**Comment by AkashKumar7902** [comment]
the go arm binary is not running on a amd machine. I am looking for a workaround.

**Comment by charankamarapu** [comment]
try with the example  given in issue to use `GOARCH=arm`. If possible include both amd and arm in one file only for each action.

**Comment by AkashKumar7902** [comment]
@charankamarapu I think i couldn't get you. In [766da40](https://github.com/keploy/keploy/pull/988/commits/766da403d4c14ab4d67d3f43e7e2c8d13713859f), i have added architecture=arm which sets the go env variable GOARCH to arm. I believe you meant something else.

**Comment by charankamarapu** [comment]
You can refer this https://opensource.com/article/21/1/go-cross-compiling. If you have already referred this and made changes and it didn't work please inform me.

File: .github/workflows/sample-run.yml [hunk 1]
```diff
@@ -21,31 +21,19 @@ jobs:
         run: |
           go build -v ./...
 
-      # - name: Setup Java
-      #   uses: actions/setup-java@v3
-      #   with:
-      #     java-version: "11"
-      #     distribution: "temurin"
+  build-arm:
```
Comment (charankamarapu) [inline .github/workflows/sample-run.yml:24]: also you dont have to create everything again just another section 
```

- name: Build Keploy ARM
   run : ........
```

Comment (AkashKumar7902) [reply .github/workflows/sample-run.yml]: Done

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -12,47 +12,14 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
-
-#      - name: Check commit messages
-#        uses: wagoid/commitlint-github-action@v4
-
       - name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v4
```
Comment (charankamarapu) [inline .github/workflows/main.yml:16]: Is this change need or relevant..?

Comment (charankamarapu) [reply .github/workflows/main.yml:16]: please revert if not

Comment (charankamarapu) [reply .github/workflows/main.yml:16]: everything else looks good.

**Comment by charankamarapu** [comment]
you forgot to sign the commit

File: .github/workflows/main.yml (hunk 2 above)
Comment (AkashKumar7902) [reply .github/workflows/main.yml]: @charankamarapu done

**Review by charankamarapu — APPROVED** [review]
ship it !

**Pull request merged**